Assets/Scripts/Interactable.cs

```diff
@@ -0,0 +1,44 @@
+﻿using UnityEngine;
+
+public class Interactable : MonoBehaviour {
+
+	public float radius = 3f;										//how close to interact - default 3 units
+
+	public Transform interactionTransform;
+	bool isFocus = false;
+	Transform player;
+	bool hasInteracted = false;
+
+	void Update() {
+		if(isFocus && !hasInteracted) {
+			float distance = Vector3.Distance(player.position, interactionTransform.position);
+			if(distance >= radius) {
```

Comment:
dont do this programatically, use hitboxes instead. this whole check can be simplified to a couple simple steps 
```
1) user clicks thing -> set destination to the target
2) hit boxes fire no matter what, then determine if we need to move or not
3) profit with less code
```